src/components/issues/Card.tsx

```diff
@@ -5,21 +5,45 @@ import { toast, ToastTypes } from '@/helperFunctions/toast';
 
 import fetch from '@/helperFunctions/fetch';
 import { IssueCardProps } from '@/interfaces/issueProps.type';
-import { TASKS_URL } from '../../constants/url';
+import { TASKS_URL, TASK_REQUEST_URL } from '../../constants/url';
 import useUserData from '@/hooks/useUserData';
 import ActionForm from './ActionForm';
 import { useRouter } from 'next/router';
+import TaskRequestForm from './TaskRequestForm';
+import Modal from '../Modal';
+import { useUpdateTaskMutation } from '@/app/services/tasksApi';
+import { REQUEST_TABS } from './constants';
+import { TASK_REQUEST_TYPES } from '@/constants/tasks';
+import { FEATURE } from '@/constants/task-type';
+import { AVAILABLE } from '@/constants/task-status';
+import { TBD } from '@/constants/constants';
 const { SUCCESS, ERROR } = ToastTypes;
+type TaskData = {
+    assignee?: string;
+    endsOn?: number;
+    startedOn?: number;
+    status?: string;
+};
 
 const Card: FC<IssueCardProps> = ({ issue }) => {
     const date = new Date(issue.created_at).toDateString();
     const [taskExists, setTaskExists] = useState(issue.taskExists ?? false);
     const [isLoading, setIsLoading] = useState(false);
     const router = useRouter();
     const devMode = router.query.dev === 'true' ? true : false;
-    const { isUserAuthorized } = useUserData();
+    const { data: userData, isUserAuthorized } = useUserData();
     const [taskId, setTaskId] = useState(issue.taskId);
-
+    const [isTaskModalOpen, setIsTaskModalOpen] = useState(false);
+    const [requestId, setRequestId] = useState();
+    const [assignee, setAssignee] = useState<string | undefined>();
+    const defaultTaskConversionTab = isUserAuthorized
+        ? REQUEST_TABS.TASK_CREATION
+        : REQUEST_TABS.CREATION_REQUEST;
```

Comment:
- should we bring this below all the hooks so that all hooks can be together for better readability?

Reply:
Done.